### PR TITLE
Refactor Windows interface_addresses table to use win32 API and report accurate netmasks.

### PR DESCRIPTION
**Pull request by ryanheffernan**
This change does the following:

- Updates the interface_addresses table to use the win32 APIs instead of WMI to query interfaces; we have found this to be a more reliable way to access data on windows.
- Updates the 'mask' column to be consistent with other platforms by reporting a subnet mask vs a prefix length
- adds a 'type' field to the table to differentiate between DHCP vs manual vs auto-assigned addresses (i.e. slaac).

### CODE REVIEW

**Comment by osqueryer** [comment]
:-1: The commit a4e9ea7e2051c01cefd3cd18d2891c8cc50ffa8b (Job results: [874](https://jenkins.osquery.io/job/osqueryPullRequestOSX/874/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit a4e9ea7e2051c01cefd3cd18d2891c8cc50ffa8b (Job results: [4727](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4727/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit b1b69f951649a9d736d4a19bb9156b021832a53e (Job results: [4729](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4729/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit b1b69f951649a9d736d4a19bb9156b021832a53e (Job results: [4730](https://jenkins.osquery.io/job/osqueryPullRequestLinux/4730/)) failed one or more tests (**Linux**).

File: osquery/tables/networking/windows/interfaces.cpp [hunk 1]
```diff
@@ -80,32 +89,98 @@ QueryData genInterfaceDetails(QueryContext& context) {
...
+        r["type"] = "unknown";
       }
+
+      if (ipaddr->Address.lpSockaddr->sa_family == AF_INET) {
+        ULONG mask;
+        ConvertLengthToIpv4Mask(ipaddr->OnLinkPrefixLength, &mask);
+        in_addr maskAddr;
+        maskAddr.s_addr = mask;
+
+        char addrBuff[INET_ADDRSTRLEN];
```
Comment (theopolis) [inline osquery/tables/networking/windows/interfaces.cpp]: You can 0 this with `char addrBuff[INET_ADDRSTRLEN] = {0};`

File: osquery/tables/networking/windows/interfaces.cpp [hunk 2]
```diff
@@ -80,32 +89,98 @@ QueryData genInterfaceDetails(QueryContext& context) {
...
+                  addrBuff,
+                  INET_ADDRSTRLEN);
+        r["address"] = addrBuff;
+      } else if (ipaddr->Address.lpSockaddr->sa_family == AF_INET6) {
+        in6_addr netmask;
+        memset(&netmask, 0x0, sizeof(netmask));
+        for (long i = ipaddr->OnLinkPrefixLength, j = 0; i > 0; i -= 8, ++j)
+          netmask.s6_addr[j] = i >= 8 ? 0xff : (ULONG)((0xffU << (8 - i)));
+
+        char addrBuff[INET6_ADDRSTRLEN];
```
Comment (theopolis) [inline osquery/tables/networking/windows/interfaces.cpp]: same here

File: osquery/tables/networking/windows/interfaces.cpp [hunk 3]
```diff
@@ -80,32 +89,98 @@ QueryData genInterfaceDetails(QueryContext& context) {
...
   }
-
-  return results_data;
-}
+  return results;
 }
-}
\ No newline at end of file
+} // namespace tables
+} // namespace osquery
```
Comment (theopolis) [inline osquery/tables/networking/windows/interfaces.cpp]: nit, newline if you are editing again

**Comment by osqueryer** [comment]
:no_good: The commit 6e6f7c71b5e162c67085ae432e2d7844906ba7eb (Job results: [1596](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1596/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:no_good: The commit ba5e4da68fd9148f67324865c7d0457277431914 (Job results: [1598](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1598/)) failed the code audit or documentation test.

**Pull request merged**